tests/test_kitti_format.py

```diff
@@ -0,0 +1,438 @@
+                        attributes={'is_crowd': False}, id=2,
+                        label=self._label('label_2')),
+                ])
+
+            def categories(self):
+                label_map = OrderedDict()
+                label_map['background'] = None
+                label_map['Label_1'] = None
+                label_map['label_2'] = None
+                return make_kitti_categories(label_map)
+
+        with TestDir() as test_dir:
+            self._test_save_and_load(SrcExtractor(),
+                partial(KittiConverter.convert, label_map='source',
+                save_images=True), test_dir, target_dataset=DstExtractor())
+
+    def test_dataset_with_source_labelmap_defined(self):
+        class SrcExtractor(TestExtractorBase):
+            def __iter__(self):
+                yield DatasetItem(id=1, image=np.ones((1, 5, 3)), annotations=[
+                    Mask(image=np.array([[1, 0, 0, 1, 1]]), label=1, id=1,
+                        attributes={'is_crowd': False}),
+                    Mask(image=np.array([[0, 1, 1, 0, 0]]), label=2, id=2,
+                        attributes={'is_crowd': False}),
+                ])
+
+            def categories(self):
+                label_map = OrderedDict()
+                label_map['background'] = (0, 0, 0)
+                label_map['label_1'] = (1, 2, 3)
+                label_map['label_2'] = (3, 2, 1)
+                return make_kitti_categories(label_map)
+
+        class DstExtractor(TestExtractorBase):
+            def __iter__(self):
+                yield DatasetItem(id=1, image=np.ones((1, 5, 3)), annotations=[
+                    Mask(image=np.array([[1, 0, 0, 1, 1]]),
+                        attributes={'is_crowd': False}, id=1,
+                        label=self._label('label_1')),
+                    Mask(image=np.array([[0, 1, 1, 0, 0]]),
+                        attributes={'is_crowd': False}, id=2,
+                        label=self._label('label_2')),
+                ])
+
+            def categories(self):
+                label_map = OrderedDict()
+                label_map['background'] = (0, 0, 0)
+                label_map['label_1'] = (1, 2, 3)
+                label_map['label_2'] = (3, 2, 1)
+                return make_kitti_categories(label_map)
+
+        with TestDir() as test_dir:
+            self._test_save_and_load(SrcExtractor(),
+                partial(KittiConverter.convert, label_map='source',
+                save_images=True), test_dir, target_dataset=DstExtractor())
+
+    def test_can_save_and_load_image_with_arbitrary_extension(self):
+        class TestExtractor(TestExtractorBase):
+            def __iter__(self):
+                return iter([
+                    DatasetItem(id='q/1', image=Image(path='q/1.JPEG',
+                        data=np.zeros((4, 3, 3)))),
+
+                    DatasetItem(id='a/b/c/2', image=Image(
+                             path='a/b/c/2.bmp', data=np.ones((1, 5, 3))
```

Comment:
Check images are saved with original extensions

Reply:
Done

Reply:
Maybe, its better to add an `osp.isfile()` check below?

Reply:
Added